db/migrate/20150213091223_add_listing_fields_to_transaction.rb

```diff
@@ -0,0 +1,8 @@
+class AddListingFieldsToTransaction < ActiveRecord::Migration
+  def change
+    add_column :transactions, :listing_title, :string, after: :listing_quantity
+    add_column :transactions, :listing_author_id, :string, after: :listing_quantity
+    add_column :transactions, :unit_price_cents, :integer, after: :listing_title
+    add_column :transactions, :unit_price_currency, :string, limit: 8, after: :unit_price_cents
```

Comment:
Yeah, I don't see the point of extra syntax to learn, write and read for a simple thing like this. Also not sure if you can control the varchar length like this with it.